worlds/yoshisisland/Rom.py

```diff
@@ -0,0 +1,1221 @@
+        rom.write_bytes(0x010045, ([world.leader_color]))
+        rom.write_bytes(0x01003D, bytearray(world.color_order))
+        rom.write_bytes(0x010043, ([world.leader_color]))
+    elif world.options.yoshi_colors.value in {2, 3}:
+        rom.write_bytes(0x010000, bytearray(world.level_colors))
+
+    if world.options.minigame_checks.value in {2, 3}:
+        bonus_checks(rom)
+
+    if world.options.minigame_checks.value in {1, 3}:
+        bandit_checks(rom)
+
+    rom.write_bytes(0x00BF2C, bytearray(world.world_bonus))
+
+    if world.options.softlock_prevention == 1:
+        rom.write_bytes(0x00C18F, bytearray([0x5C, 0x58, 0xFB, 0x0B])) #R + X Code
+
+    if world.options.bowser_door_mode.value != 0:
+        rom.write_bytes(0x07891F, bytearray(world.castle_door)) #1 Entry
+        rom.write_bytes(0x078923, bytearray(world.castle_door)) #2 Entry
+        rom.write_bytes(0x078927, bytearray(world.castle_door)) #3 Entry
+        rom.write_bytes(0x07892B, bytearray(world.castle_door)) #4 Entry
+
+    if world.options.bowser_door_mode.value == 5:
+        rom.write_bytes(0x0AF517, bytearray([0xC6, 0x07, 0x7A, 0x00])) #Door 2
+        rom.write_bytes(0x0AF6B7, bytearray([0xCD, 0x05, 0x5B, 0x00])) #Door 3
+        rom.write_bytes(0x0AF8F2, bytearray([0xD3, 0x00, 0x77, 0x06])) #Door 4
+
+    if world.options.goal == 1:
+        rom.write_bytes(0x1153F6, bytearray([0x16, 0x28, 0x10, 0x0C, 0x10, 0x4E, 0x1E, 0x10, 0x08, 0x04, 0x08, 0x24, 0x36, 0x82, 0x83, 0x83, 0x34, 0x84, 0x85, 0x85])) #Luigi piece clear text
+        rom.write_bytes(0x06FC86, bytearray([0xFF])) #Boss clear goal = 255, renders bowser inaccessible
+
+    from Main import __version__
+    rom.name = bytearray(f'YOSHIAP{__version__.replace(".", "")[0:3]}_{player}_{multiworld.seed:11}\0', 'utf8')[:15]
+    rom.name.extend([0] * (15 - len(rom.name)))
+    rom.write_bytes(0x007FC0, rom.name)
+
+class YIDeltaPatch(APDeltaPatch):
+    hash = USHASH
+    game: str = "Yoshi's Island"
+    patch_file_ending = ".apyi"
+
+    @classmethod
+    def get_source_data(cls) -> bytes:
+        return get_base_rom_bytes()
+
+
+
+def get_base_rom_bytes(file_name: str = "") -> bytes:
+    base_rom_bytes = getattr(get_base_rom_bytes, "base_rom_bytes", None)
+    if not base_rom_bytes:
+        file_name = get_base_rom_path(file_name)
+        base_rom_bytes = bytes(Utils.read_snes_rom(open(file_name, "rb")))
+
+        basemd5 = hashlib.md5()
+        basemd5.update(base_rom_bytes)
+        if USHASH != basemd5.hexdigest():
+            raise Exception('Supplied Base Rom does not match known MD5 for US(1.0) release. '
+                            'Get the correct game and version, then dump it')
+        get_base_rom_bytes.base_rom_bytes = base_rom_bytes
+    return base_rom_bytes
+
+def get_base_rom_path(file_name: str = "") -> str:
+    options: Utils.OptionsType = Utils.get_options()
+    if not file_name:
+        file_name = options["yoshisisland_options"]["rom_file"]

```

Comment:
```
from settings import get_settings

```
```suggestion
    if not file_name:
        file_name = get_settings()["yoshisisland_options"]["rom_file"]

```
Utils' get_options is deprecated. Use the settings one instead.